include/bluetooth/mesh/scene_srv.rst

```diff
@@ -34,39 +34,58 @@ It's up to the individual model implementation to correctly serialize and deseri
 
-All models that extend these models will automatically become part of the scenes as well.
-
 Adding scene data for a model
 *****************************
 
 Every model that needs to store its state in scenes must register a :cpp:type:`bt_mesh_scene_entry` structure containing callbacks for storing and recalling the scene data.
-The scene entry must be registered with a valid :cpp:type:`bt_mesh_scene_entry_type` structure, which defines the store and recall behavior of the scene entry.
+The scene entry must be declared using :c:macro:`BT_MESH_SCENE_ENTRY_SIG` for SIG defined models (e.g. from the model specification), or :c:macro:`BT_MESH_SCENE_ENTRY_VND` for Vendor specific models.
+These macros will allocate the structure in flash, and make them known to the Scene Server model.
 
-For instance, the Generic OnOff Server model defines its scene entry by adding a `bt_mesh_scene_entry` structure to every :cpp:type:`bt_mesh_onoff_srv`, and registering it with a common type:
+For instance, the Generic OnOff Server model defines its scene entry by declaring a :cpp:type:`bt_mesh_scene_entry` structure with the Generic OnOff Server model ID and a set of callbacks:
 
 .. literalinclude:: ../../../subsys/bluetooth/mesh/gen_onoff_srv.c
    :language: c
    :start-after: include_startingpoint_scene_srv_rst_1
    :end-before: include_endpoint_scene_srv_rst_1
 
-The Generic OnOff Server model only needs to store its OnOff state in the scene, so it always stores and recalls a single byte.
-The ``restore`` call contains a transition time pointer, which should be taken into account when implementing the model behavior.
+The Generic OnOff Server model only needs to store its OnOff state in the scene, so it always stores and recalls a single byte, as reflected in the :c:member:`bt_mesh_scene_entry.maxlen` member of the :cpp:type:`bt_mesh_scene_entry`.
+
+Storing and recalling scene data
+********************************
+
+When storing and recalling a scene, the Scene Server will go through each model in the composition data elements it represents, and check if this model has a scene entry.
+
+.. note::
+   Models that are extended by other models will neither store nor recall scene data, and will instead rely on the extending models to maintain this data for them.
+
+If a model has a scene entry, and is not extended by other models, the Scene server will use the callbacks in the scene entry to either store or recall the model's scene data.
+
+As the Scene entries are registered per model ID, all models with that ID will be requested to store its scene data, unless if they're extended.
+To skip a specific model instance, return a negative value from the ``store`` callback.
 
 If a scene store operation is requested in the middle of a transition between two states, the model should commit the target state to the scene data.
 
+The ``restore`` call contains a transition time pointer, which should be taken into account when implementing the model behavior.
+
```

Comment:
May I suggest something like this?
```suggestion
Storing and recalling scene data
********************************
When storing and recalling a scene, the Scene Server will go through each model in the composition data elements it represents, and check if there is a scene entry for this model ID.
The Scene server will use the callbacks in the scene entry to either store or recall the model's scene data for each model whose model ID is specified in the scene entry.

If you want to skip a specific model instance, return a negative value from the ``store`` callback.

If a model is :ref:`extended <zephyr:bluetooth_mesh_access#extended-models>` by other models, it will neither be stored nor recalled with the scene regardless of whether there is a scene entry for this model ID.

If a scene store operation is requested in the middle of a transition between two states, the model should commit the target state to the scene data.

The ``recall`` callback contains a transition time pointer, which should be taken into account when implementing the model behavior.
```

Reply:
Either way works for me :)
Just two things:
- Scene server -> Scene Server
- If a model is :ref:`extended <zephyr:bluetooth_mesh_access#extended-models>` by other models, it will neither be stored nor recalled with the scene, even if there is a scene entry for this model ID.